### PR TITLE
FIX: update documentation to use cleaner furo theme and add markdown support via myst

### LINKED ISSUE #276: Add myst support to documentation to allow for markdown
If we add myst parser to documentation we can then use markdown with all of the features of rst. we should update it as it still also allows us to use rst if we wish. 

Also embed CONTRIBUTING file into the documentation so people can find it in both places. And make a more welcoming landing page for documentation.

### PR DESCRIPTION
**Pull request by lwasser**
closes #276 
This pr does a bunch of cleanup things

* It adds support for myst which allows us to include markdown in our docs for easier contributing! 
* It cleans up the landing page of the docs to make it easier for users to find content
* It embeds the CONTRIBUTING.md file into the documentation for easier discovery
* It just cleans up extra stuff from the conf.py file that we don't need! 
* I finessed the documentation styles so its just a bit cleaner as well :) 

In case you don't want to build locally i've added some screenshots of what the new theme and landing page look like! 

# Documentation landing page
(I might make the grid cards grey or add more of a drop shadow at some point) 

Screen Shot 2022-11-26 at 11 37 22 AM

## Landing page lower down 
Screen Shot 2022-11-26 at 11 37 29 AM

## New sidebar navigation 

so nav is not split between top and side which i think is confusing .
Screen Shot 2022-11-26 at 11 37 11 AM

### CODE REVIEW

File: docs/conf.py
Comment (lwasser) [inline docs/conf.py:94]: This just adds the custom css to clean up the look of our docs!

File: docs/contributing/how-to-contribute.md
Comment (lwasser) [inline docs/contributing/how-to-contribute.md:3]: this is the embed to contributing.md is in the documentation too!

File: docs/index.md
Comment (lwasser) [inline docs/index.md:3]: this makes the landing page prettier / easier for a user to navigate

File: docs/index.md
Comment (lwasser) [inline docs/index.md:87]: This is the hidden navigation that the user sees on all pages on the left hand navigation bar

Screen Shot 2022-11-26 at 11 37 11 AM

File: docs/index.md
Comment (lwasser) [reply docs/index.md:3]: Screen Shot 2022-11-26 at 11 37 22 AM

**Comment by lwasser** [comment]
The last PR will add a development guide. i see docs are failing so let me look into that now.

**Comment by lwasser** [comment]
ok all passing now and I squashed down commits! ready for review!

File: docs/conf.py
Comment (jsamoocha) [reply docs/conf.py:94]: Ok, I think the .css sets a fixed font color that creates the issue in dark mode. See my inline comment.

File: docs/_static/stravalib.css
Comment (jsamoocha) [inline docs/_static/stravalib.css:6]: Maybe we should not hard-code colors here? These seem to lead to readability issues in dark mode. Maybe it's better (and easier to maintain) to let the theme decide about colors and/or font sizes to prevent any potential accessibility issues?

**Review by jsamoocha — COMMENTED** [review]
Thanks for this contribution. The documentation starts to look good! Being able to use markdown is a great improvement.

There is an accessibility issue with the custom .css, please see my other comments.

There is also a problem rendering the changelog:

Notification_Center

I'm also missing the latest entries to the changelog. Can it be that it's not correctly included/embedded?

File: docs/conf.py
Comment (lwasser) [reply docs/conf.py:94]: @jsamoocha yes great catch. i hadn't tested on dark mode. I just pushed a fix that removes all color settings from the CSS. i keep forgetting about dark mode options in these themes. 

Screen Shot 2022-11-29 at 8 24 04 AM

File: changelog.md
Comment (lwasser) [inline changelog.md:104]: removed rst that i must have missed in that other big pr

File: docs/conf.py
Comment (jsamoocha) [reply docs/conf.py:94]: It was pure coincidence that I found that dark mode issue - just decided to try `make -C docs serve` after sunset 😁

File: docs/contributing/documentation-changelog.md
Comment (jsamoocha) [reply docs/contributing/documentation-changelog.md:3]: I pulled your recent changes locally, but the changelog seems to have disappeared completely. Am I missing something?

Stravalib_Docs

Comment (lwasser) [reply docs/contributing/documentation-changelog.md:3]: oh gosh. Let me look again locally. I thought i had fixed that. 

There is a "feature" / "bug" where if there are two h1 tags on a page, it won't render in the TOC.  BUT there is no header in that page at all purposefully to address that bug. Let me see if it needs a different approach. thank you again for catching all of these issues. goodness.

File: docs/index.md
Comment (lwasser) [inline docs/index.md:112]: Oh i see the issue @jsamoocha i bet my local build needed a clean and update. that link needs to go to:

contributing/documentation-changelog

This is a leah-error for sure.

Comment (lwasser) [inline docs/index.md:112]:
```suggestion
Change Log <contributing/documentation-changelog>
```

File: docs/index.md
Comment (lwasser) [inline docs/index.md:112]: I will test this locally @jsamoocha in hopes of saving you time :) my goodness

**Comment by lwasser** [comment]
ok i built this locally. funny enough what happened was i fixed the index file and the docs were building fine locally. but i never pushed or committed the change to GitHub. it should be clean now @jsamoocha i hope!!  ive looked at it in dark and regular mode too. i see the changelog in the side bar. hopefully it works for you as well.

**Comment by lwasser** [comment]
there is one quirk in the PR that i'm seeing that should get fixed with the other open  draft PR https://github.com/stravalib/stravalib/pull/278 it's failing because it needs this PR with the requirements update. the full side bar should work and be cleaner with that PR in place so once we merge this i'll work on that one! (assuming the changelog works now ) and styles are readable for you!

**Comment by jsamoocha** [comment]
Ok, almost there. I noticed a broken link (to the newly added code of conduct). It then occurred to me that the linkcheck target is not part of the doc build anymore and therefore didn't catch this. When adding and building with linkcheck locally, there are 2 more broken links in the docs.

Maybe we should make sure that all doc build targets depend on linkcheck?

**Comment by lwasser** [comment]
@jsamoocha i'm happy to add that and am working on it locally before i start my day job :)  is see those errors now. i'll add this either at lunch time or this evening as i want to rethink the makefile just a little bit with this change. I think there will be one broken link on the contributing file that is to a file in https://github.com/stravalib/stravalib/pull/278 that PR was built upon this pr and depends on this pr. I could simply remove the link and will just do that in this pr and re-add it in the draft pr. I just didn't want to have a GINORMOUS pr for you to review it's so much harder. here we've been able to go back and forth on a lot of things that i think were easier to catch with a smaller pr. but i see there are some other broken links too so i'll work on those now locally and add link check back to the build! great catch!

**Comment by lwasser** [comment]
this is NOT READY to review - will fix links later today. but hopefully the CI runs as we want it to.

File: .github/workflows/build-docs.yml
Comment (lwasser) [inline .github/workflows/build-docs.yml:39]: i added linkcheck to the vanilla make docs call here.

File: CONTRIBUTING.md
Comment (lwasser) [inline CONTRIBUTING.md:25]: i may embed the COC  into the docs for cleaner reading in another PR but not here.

File: docs/Makefile
Comment (lwasser) [inline docs/Makefile:44]: link check now will produce an output file output.txt with all broken things ... much easier to read as well vs looking at the CLI output of sphinx/linkcheck

File: docs/Makefile
Comment (lwasser) [inline docs/Makefile:35]: so remove that dir in case you wish to do a full clean of docs

**Comment by lwasser** [comment]
PHEW! ok this is all ✅ so we're good for another review. it looks ok locally. i've fixed a bunch of broken and redirected links. And updated the makefile accordingly. also found a changelog item that i must have added that hasn't happened yet :) removed that even tho the psychic in me knows it will happen in #278 once it's merged :) and i fixed a toctree bug that was making a doc page render in the side bar weirdly too!

**Review by jsamoocha — APPROVED** [review]
LGTM, great work!

**Comment by lwasser** [comment]
sweet merging. thanks for the careful reviews @jsamoocha !!

**Pull request merged**
